tests/evm/test_sha3.py

```diff
@@ -0,0 +1,139 @@
+import pytest
+
+from zkevm_specs.evm import (
+    Block,
+    Bytecode,
+    CopyCircuit,
+    CopyDataTypeTag,
+    ExecutionState,
+    KeccakCircuit,
+    Opcode,
+    RWDictionary,
+    StepState,
+    Tables,
+    verify_steps,
+)
+from zkevm_specs.copy_circuit import verify_copy_table
+from zkevm_specs.util import (
+    keccak256,
+    memory_expansion,
+    memory_word_size,
+    rand_bytes,
+    rand_fq,
+    FQ,
+    GAS_COST_COPY_SHA3,
+    MEMORY_EXPANSION_LINEAR_COEFF_SHA3,
+    RLC,
+    U64,
+)
+
+
+CALL_ID = 1
+TESTING_DATA = ((0x20, 0x40),)
+
+
+@pytest.mark.parametrize("offset, length", TESTING_DATA)
+def test_sha3(offset: U64, length: U64):
+    randomness = rand_fq()
+
+    offset_rlc = RLC(offset, randomness)
+    length_rlc = RLC(length, randomness)
+
+    # divide rand memory into chunks of 32 which we will push and mstore.
+    memory_snapshot = rand_bytes(offset + length)
+    memory_chunks = list()
+    for i in range(0, len(memory_snapshot), 32):
+        memory_chunks.append(memory_snapshot[i : i + 32])
+    src_data = dict(
+        [
+            (i, memory_snapshot[i] if i < len(memory_snapshot) else 0)
+            for i in range(offset, offset + length)
+        ]
+    )
+
+    bytecode = Bytecode()
+    for i, chunk in enumerate(memory_chunks):
+        bytecode.push(32 * i, n_bytes=32).push(chunk, n_bytes=32).mstore()
+    bytecode.push(offset_rlc, n_bytes=32).push(length_rlc, n_bytes=32).sha3().stop()
```

Comment:
Although these opcodes are not tested in this test, I think it should be `offset` and `length` instead of their RLC versions.

Reply:
oops, that's a silly error. Fixing it now.